chartfx-dataset/src/test/java/de/gsi/dataset/serializer/benchmark/SerialiserAssumptionsBenchmark.java

```diff
@@ -0,0 +1,271 @@
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public int functionWithSingleArgument(Blackhole blackhole, final MyData data) {
+        return f3(blackhole, data.a);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public int[] functionWithVarargsArrayArgument(Blackhole blackhole, final MyData data) {
+        return f1(blackhole, data.dim);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public int[] functionWithVarargsMultiArguments(Blackhole blackhole, final MyData data) {
+        return f1(blackhole, data.a, data.b, data.c);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public int[] functionWithVarargsSingleArgument(Blackhole blackhole, final MyData data) {
+        return f1(blackhole, data.a);
+    }
+
+    @Setup()
+    public void initialize() {
+        // add variables to initialise here
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public String stringAllocationWithCharsetASCII(final MyData data) {
+        return new String(data.byteASCII, 0, data.byteASCII.length, StandardCharsets.US_ASCII);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public String stringAllocationWithCharsetISO8859(final MyData data) {
+        return new String(data.byteISO8859, 0, data.byteISO8859.length, StandardCharsets.ISO_8859_1);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public String stringAllocationWithCharsetUTF8(final MyData data) {
+        return new String(data.byteISO8859, 0, data.byteISO8859.length, StandardCharsets.UTF_8);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public String stringAllocationWithCharsetUTF8_UTF8(final MyData data) {
+        return new String(data.byteUTF8, 0, data.byteUTF8.length, StandardCharsets.UTF_8);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public String stringAllocationWithOutCharsetASCII(final MyData data) {
+        return new String(data.byteASCII, 0, data.byteASCII.length);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public String stringAllocationWithOutCharsetISO8859(final MyData data) {
+        return new String(data.byteISO8859, 0, data.byteISO8859.length);
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public String stringAllocationWithOutCharsetISO8859_V2(final MyData data) {
+        return new String(data.byteISO8859, 0, 0, data.byteISO8859.length); // NOPMD NOSONAR fast implementation
+    }
+
+    @Benchmark
+    @Warmup(iterations = 1)
+    @Fork(value = 2, warmups = 2)
+    public String stringAllocationWithOutCharsetISO8859_V3(final MyData data) {
```

Comment:
Codacy found an issue: [The instance method name 'stringAllocationWithOutCharsetISO8859_V3' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5905529)